### ISSUE TITLE
Duda acerca del saldo del chip del modem

### ISSUE THREAD

**Issue by Nnino** [edwinspire/usaga#55]
Te quería consultar si es que se encuentra dentro de los planes hacer un módulo que permita conocer el saldo (dinero) que contiene el chip?

**Comment by edwinspire** [comment]
Pues no está planificado, al menos aca en Ecuador no hay forma de 
saberlo via sms u otra forma.
Se puede conocer el saldo uninamente haciendo una llamada telefónica.
No sabria como implementarlo si no hay esa funcion disponible.

**Comment by Nnino** [comment]
Entiendo. 

Lo que pasa es que acá en Chile, si uno envía un SMS al número telefónico 300 con el texto S, la compañía devuelve un SMS con el saldo del chip. 
Entonces te quería preguntar si había alguna posibilidad que pudieras hacer un módulo que enviara un mensaje con las características mencionadas, y que el mensaje recibido de ese número (el que contiene el saldo) se muestre en pantalla (de la misma forma que muestras los mensajes entrantes, pero en lugar de mostrarlos todos, sólo mostrar el mensaje más reciente proveniente del número 300).

**Comment by edwinspire** [comment]
Es posible hacerlo.
Enviarme exactamente como llega el mensaje proveniente de ese número 
para crear un módulo adicional.

**Comment by Nnino** [comment]
Actualmente no tengo la información técnica de como llega este mensaje, pero por algún motivo ayer quedó registrado en el sistema un mensaje proveniente de la compañía en el módulo de Mensajes Entrantes. Apenas tenga acceso al sistema te enviaré los detalles de este registro. Si necesitas alguna información adicional puedo buscarla.

Más o menos el funcionamiento es el que te expliqué arriba, uno envía un SMS al número 300 con el texto S y la compañía devuelve un mensaje de texto con el saldo. 
Ayer hice un par de pruebas, y probé el envío del SMS desde un celular y desde el modem por medio de comandos AT (esto lo hice con el mismo chip que usamos habitualmente para el modem). Desde el celular se recibió sin problemas el mensaje de la compañía, pero con los comandos AT en modo texto no hubo respuesta, por lo que supongo que el modo PDU es el que debería funcionar.

Un detalle importante de esta forma de saber el saldo, es que tiene un costo adicional sobre el saldo del chip. Por lo que la idea es que no se muestre el saldo inmediatamente después de ingresar al módulo, sino que exista algún botón que desencadene la acción y se le advierta a la persona que esta operación tiene un costo en el saldo.

**Comment by Nnino** [comment]
![captura](https://f.cloud.github.com/assets/3172011/678252/d2c6d968-d92f-11e2-8529-3755e52d55f0.png)

Ahí esta la captura del mensaje con el saldo enviado por la compañía . No se muestra el saldo ahí porque el chip no tenía y la compañía me estaba avisando de ello.

**Comment by edwinspire** [comment]
Bien, enviarme una captura tambien del mensaje cuando tiene saldo para 
ver cual es el formato empleado y aplicar alguna expresión regular para 
obtener el dato.

**Comment by Nnino** [comment]
Ok, trataré de conseguir el registro lo antes posible.

**Comment by Nnino** [comment]
No me ha sido posible volver a conseguir que aparezca el SMS de la compañía en el sistema. Aún así el SMS que envía la compañía tiene el siguiente texto: "Tu saldo es $5000, expira 23-9-2013. Saldo numero preferido: 400 minutos."

**Comment by edwinspire** [comment]
Ok.
El modulo de envio de mensajes de texto esta parcialmente funcionando, 
probar enviar el sms de solicitud de saldo para ver si se registra en el 
sistema.

**Comment by Nnino** [comment]
Los cambios están en la versión develop?

**Comment by edwinspire** [comment]
Si porque aun no esta terminado.

**Comment by Nnino** [comment]
Ok, estuve probándolo y envía SMS a celulares normales, pero no he recibido el SMS de la compañía.

**Comment by Nnino** [comment]
Ah... en la versión develop hay un error en el módulo SMS Salientes.

**Comment by edwinspire** [comment]
Que error aparece?

**Comment by Nnino** [comment]
Solamente se ve la tabla de SMS salientes y nada mas. Cuando se aprieta el botón "Enviar" no funciona. Y en el navegador se muestran los siguientes errores:
Failed to load resource: the server responded with a status of 404 (NOT FOUND)   http://localhost:8081/lib/custom/_usms_smsout_process/_usms_smsout_process.js
Error  dojo.js:1108

**Comment by edwinspire** [comment]
Solucionado

**Comment by Nnino** [comment]
Ok. He seguido probando pero aunque el SMS se envía y queda en el historial de SMS enviados, no existe respuesta por parte de la compañía. Hay alguna configuración en especial que hay que hacerle a este módulo antes de enviar un SMS?

**Comment by edwinspire** [comment]
Existe alguna otra forma de averiguar el saldo? pero debe ser peor sms.

**Comment by Nnino** [comment]
No, esa es la única forma que ofrece esta compañía. Las demás formas son por medio de llamada, marcando un número en el celular o entrando a la sucursal virtual de la compañía.

**Comment by cttemux** [comment]
> Es posible hacerlo.
> Enviarme exactamente como llega el mensaje proveniente de ese número 
> para crear un módulo adicional.
> 

Nunca aja sido por SMS la comunicación solo Telegram y WhatsApp

**Comment by cttemux** [comment]
Soy de México y está persona me me encomendo un trabajo y jamás me pagó.

**Comment by edwinspire** [comment]
A que persona se refiere?
